### ISSUE TITLE
Add new crate for types definitions only

### ISSUE THREAD

**Issue by mattsse** [gakonst/ethers-rs#2237]
ethers-core has a lot of types that would be useful as standalone package,

for example all tracing-related types (https://github.com/paradigmxyz/reth/issues/1617)

It would also make it easier to have a crate just for tracing type defs that can be reused easily.
this way ruint/primitive types can be feature gated as well.

**Comment by gakonst** [comment]
Good idea - @prestwich and I have been talking about trying to use as many reth types as possible in ethers, ideally, using Reth as the source of truth

**Comment by prestwich** [comment]
the reth-primitives crate is large, unstable, and unpublished. so we can't really use it as a source of truth
